### PR TITLE
move sensor_reporter to YAML configuration

### PR DESCRIPTION
**Pull request by DanielDecker**
Convert sensor_reporter from ini config file to YMAL configuration.
All modules are translated now.

Currently following was tested and is working with the YAML config:
- [x] local connector
- [x] openHAB connector
- [x] MQTT connector with Retain parameter #81
- [x] exec actuator and exec sensor
- [x] RpiGpioSensor/Actuator and DhtSensor (3/3)
- [x] heartbeat
- [x] network sensors (2/2)
- [x] bluetooth sensors (3/3)
- [x] energymeter
- [x] roku address sensor
- [x] local logic gate #94 
- [x] modify parse_values to handle values for each connection

Following sensors are tested only partial since I don't have the associated equipment:
- DHTSensor
- govee sensor
- energymeter
- roku sensor

This addresses the discussion in #95

### CODE REVIEW

**Comment by rkoshak** [comment]
On thing that can be useful is to create a list of what's left to do using the "add task list" option.

image

That will help us both keep track of what's been done and what's left to do. 

- [ X ] done task
- [ ] to be done
- [ ] more stuff

I'll start looking at the changes when I get a chance but treat no comment as everything looks good. I don't want to slow you down.

**Comment by DanielDecker** [comment]
modified the GPIO devices to work with the YAML config. I tested the changes for the RpiGpioSensor and RpiGpioActuator. But since I don't have a DHT sensor I could only test for exceptions, so far it looked good.

**Comment by DanielDecker** [comment]
I'm mostly done with the conversion to the YAML config. I will work now on the local logic gate, but this won't change the core classes nor the connections.

**Comment by DanielDecker** [comment]
As mentioned above I couldn't test all modules in detail.
@rkoshak can you test these modules?
- DHTSensor
- govee sensor
- energymeter
- roku sensor

**Comment by rkoshak** [comment]
It might take a bit but I can definitely test Govee and Roku. I'll have to pull a DHT and breadboard out to test that and I never did have a way to test energymeter. 

I'm super busy at work leaving little time left over for hobbies right now.

**Comment by DanielDecker** [comment]
Ok,
for the DHT and the energymeter I changed as little as possible of the code, so the chance for a runtime exception should be small.

**Comment by DanielDecker** [comment]
@rkoshak I hop you're alright. Maybe you have some time to review this pull request now?

File: README.md [hunk 1]
```diff
@@ -137,12 +138,27 @@ Possible values are:
...
+DEFAULT:
+    TempUnit: F
+```
 
 # Release Notes
 This current version is a nearly complete rewrite of the previous version with a number of breaking changes.
 
 ## Breaking Changes
 
+- The configuration file is now in YAML syntax insted of a ini file - March 2022
```
Comment (rkoshak) [inline README.md:161]: Because of me we need to change that date.

File: gpio/README.md [hunk 2]
```diff
@@ -65,7 +73,7 @@ Additionally the Sensor can detect toggle events and report the time of the even
 ### Dependencies
 
 The user running sensor_reporter must have permission to access the GPIO pins.
-To grant the `sensorReporter` user GPIO permissions add the user to the group `gpio`:  `sudo adduser sensorReporter gpio`
+To grant the `sensorReporter` user GPIO permissions add the user to the group `gpio`:  `$ sudo adduser sensorReporter gpio`
```
Comment (rkoshak) [inline gpio/README.md:76]: I never added the $ to commands like this to make it easier to copy and paste. I've seen it done either way in other repos. Maybe it makes sense to make it a separate code block which I think will a copy icon.

File: gpio/README.md [hunk 3]
```diff
@@ -164,83 +212,93 @@ $ sudo pip3 install RPI.GPIO
...
-
-[Actuator0]
-Class = gpio.rpi_gpio.RpiGpioActuator
-Connection = local,openHAB
-CommandSrc = garage_light
-ToggleCommandSrc = toggle_garage_light
-Pin = 19
+### Example Config #​2
+Using a local connection to toggle an actuator, which is also connected to openHAB. 
+The actuator shows alway the correct status in openHAB, even if it is toggled locally.
```
Comment (rkoshak) [inline gpio/README.md:266]: always

**Review by rkoshak — COMMENTED** [review]
I completely dropped the ball on this. I had it in my mind I was waiting on you for something and clearly you were waiting on me. 

I won't have a chance to test this any time soon so let's just accept it and fix stuff as they come up. I will want to cut a tag to preserve the current baseline before I merge this into main so people can get back to the current version in case there's a need to. 

Over all I like how the YAML simplifies some of the code. I would have liked local logic to have been a separate PR to make review easier but I don't want to hold this up any longer.

I found just a few really minor things.

File: README.md (hunk 1 above)
Comment (DanielDecker) [reply README.md:161]: sure

File: gpio/README.md (hunk 2 above)
Comment (rkoshak) [reply gpio/README.md:76]: We should be consistent so I'd say yes.

**Comment by DanielDecker** [comment]
With the last commit I changed the readme's in accordance with your comments above.
Don't forget to cut a tag to the 'legacy code'.

**Pull request merged**